lib/checker.js

```diff
@@ -60,15 +93,22 @@ Checker.prototype.checkDirectory = function(path) {
             }
 
             return vowFs.stat(fullname).then(function(stat) {
+                var promise;
+
                 if (stat.isDirectory()) {
                     return this.checkDirectory(fullname);
                 }
 
                 if (!this._hasCorrectExtension(fullname)) {
-                    return [];
+                    if (this._configuration.getExtract() === 'never') {
+                        return [];
+                    }
+                    promise = Vow.when(this.extractFile(fullname));
```

Comment:
We could use `never` as default value for `extract` option as `JSHint` has. I believe we should do initial solution as simple as possible. Additional options (and other improvement) could be added later.

If somebody runs `jscs .` it's not a problem I guess. `never` for `extract` by default should does no any performance impact.